backstop.config.js

```diff
@@ -1,23 +1,33 @@
+const { executablePath } = require('puppeteer')
+
 const {
-  HOSTNAME = '0.0.0.0', // Default via `npm start`
-  BASE_HOST = HOSTNAME === 'docker-desktop' ? 'host.docker.internal' : HOSTNAME,
-  BASE_URL = `http://${BASE_HOST}:3000/nhsuk-frontend`
+  HEADLESS,
+  PORT = 3000,
+  BASE_HOST = `localhost:${PORT}`, // Default via `npm start`
+  BASE_URL = `http://${BASE_HOST}/nhsuk-frontend`
 } = process.env
 
 /**
  * @type {PlaywrightEngineConfig}
  */
 module.exports = {
-  asyncCaptureLimit: 5,
-  asyncCompareLimit: 50,
-  dockerCommandTemplate:
-    'docker run --rm --network=host --mount type=bind,source="{cwd}",target=/src backstopjs/backstopjs:{version} {backstopCommand} {args}',
   engine: 'playwright',
   engineOptions: {
+    args: [
+      '--deterministic-mode',
+      '--disable-skia-runtime-opts',
+      '--force-device-scale-factor=1'
+    ],
     browser: 'chromium',
-    gotoParameters: { waitUntil: 'load' }
+    chromePath: executablePath(),
+    gotoParameters: { waitUntil: 'load' },
+
+    // Allow headless mode switching using `HEADLESS=false`
+    asyncCaptureLimit: HEADLESS ? 1 : 4,
```

Comment:
I've run into two problems here.  First, the `asyncCaptureLimit` setting needs to be at the top level or it has no effect.  Second, you're taking the truth value of a string, not a boolean: if you run `HEADLESS=false npm run...` then `HEADLESS ? 1 : 4` evaluates to 1, not 4.  It needs to be:

```
  asyncCaptureLimit: HEADLESS === "false" ? 4 : 1,
```